admin/admin.php

```diff
@@ -234,9 +234,17 @@ public function add_isc_fields( $form_fields, $post ) {
 		 * While the URL to edit an attachment can also include "wp-admin.php.php", the HTTP_REFERER is then different
 		 */
 		if ( ! empty( $_SERVER['HTTP_REFERER'] )
+			 // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized,WordPress.Security.ValidatedSanitizedInput.MissingUnslash
 			&& strpos( $_SERVER['HTTP_REFERER'], 'wp-admin/post.php' ) !== false
 			// the filter allows users to force the ISC fields and Block options at the same time
 			&& ( ISC_Block_Options::enabled() && ! apply_filters( 'isc_force_block_options', false ) ) ) {
```

Comment:
**Suggestion:** Instead of using `$_SERVER['HTTP_REFERER']` directly, consider using `wp_get_referer()` function which is a safer way to get the referer URL in WordPress. [security]
```suggestion
  $referer = wp_get_referer();
  if ( ! empty( $referer )
      && strpos( $referer, 'wp-admin/post.php' ) !== false
      // the filter allows users to force the ISC fields and Block options at the same time
      && ( ISC_Block_Options::enabled() && ! apply_filters( 'isc_force_block_options', false ) ) ) {
  }
```